msktldap.cpp

```diff
@@ -126,7 +126,7 @@ LDAPMessage* ldap_get_account_attrs(const msktutil_flags* flags,
                                     const char **attrs)
 {
     std::string filter = sform("(&(|(objectCategory=Computer)"
-                               "(objectCategory=User))(sAMAccountName=%s))",
+                               "(objectCategory=User)(objectCategory=MsDS-ManagedServiceAccount))(sAMAccountName=%s))",
```

Comment:
Looks reasonable to me. Once we're at it, would it make sense to also include `msDS-GroupManagedServiceAccount` in the same vein? (And probably reformatting the lines a bit to keep them more legible.)

Reply:
Well, group managed service accounts are intended for accounts shared by several machines. Then, msktutil should not change the account password. It should use the machine credentials to retrieve the password and generate the keytab from it.

That said, yes, one can add (objectCategory=MsDS-GroupManagedServiceAccount) 

Thank you for your fast respose.

Reply:
You can look at the [Microsoft documentation](https://docs.microsoft.com/en-us/windows-server/security/group-managed-service-accounts/getting-started-with-group-managed-service-accounts)  With group managed service accounts, one sets ups the machine accounts allowed to retrieve the managed account password, with the argument `PrincipalsAllowedToRetrieveManagedPassword` of the powershell command `New-ADServiceAccount`

Reply:
Please point me to the paragraph, the resource is too large.

Reply:
Given that we've just received PR #203 to deal with group managed service accounts, let's resolve the related discussion here.